mathics/builtin/arithfns/basic.py

```diff
@@ -289,22 +289,17 @@ def format_plus(self, items, evaluation):
         "Plus[items__]"
 
         def negate(item):
-            if item.has_form("Times", 1, None):
-                if isinstance(item.leaves[0], Number):
-                    neg = -item.leaves[0]
-                    if neg.sameQ(Integer1):
-                        if len(item.leaves) == 1:
-                            return neg
-                        else:
-                            return Expression("Times", *item.leaves[1:])
+            if isinstance(item.leaves[0], Number):
```

Comment:
This already covers the `if item.has_form("Times", 1, None)`.

Reply:
> This already covers the `if item.has_form("Times", 1, None)`.

To me this sounds like you are saying the same thing but in different words: the if test was removed because it is already covered. 

The if test checks if "Head "is "Times", and item has (at least?)  one leaf. If this is not checked, how do we know that this is the case. Or how do we know this check is not needed?

Reply:
If it is `Times` we are changing to another `Times`, and if it isn't, we are creating a `Times`, so same thing.

Reply:
All of these assumptions should be written in as comments. They might be wrong and they may change at some point.